### PR TITLE
Migrate amazon provider transfer tests from `unittests` to `pytest`

### PR DESCRIPTION
**Pull request by IAL32**
Related: https://github.com/apache/airflow/pull/28039 https://github.com/apache/airflow/pull/28139 @Taragolis 

Migrate Amazon provider's transfer tests to `pytest`.

All changes are more or less straightforward:

- Get rid of unittests.TestCase class and TestCase.assert* methods
- Convert setUp* and tearDown* methods to [appropriate pytest alternative](https://docs.pytest.org/en/6.2.x/xunit_setup.html#classic-xunit-style-setup)
- Replace decorator `parameterized.expand` by `pytest.mark.parametrize`.
- Renamed `@patch` to `@mock.patch` to conform to other tests

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
some tests failing :(

File: tests/providers/amazon/aws/transfers/test_sql_to_s3.py [hunk 1]
```diff
@@ -146,11 +145,12 @@ def test_execute_json(self, mock_s3_hook, temp_mock):
...
-    @parameterized.expand(
+    @pytest.mark.parametrize(
+        "_, params",
         [
             ("with-csv", {"file_format": "csv", "null_string_result": None}),
             ("with-parquet", {"file_format": "parquet", "null_string_result": "None"}),
-        ]
+        ],
     )
     def test_fix_dtypes(self, _, params):
```
Comment (Taragolis) [inline tests/providers/amazon/aws/transfers/test_sql_to_s3.py]: With `pytest.mark.parametrize` you do not need to use unnamed variable `_` to show actual tests case.
You might use `pytest.param(..., id="some value")`. Actually this id could be use as part of run individual test

```python
    @pytest.mark.parametrize(
        "params",
        [
            pytest.param({"file_format": "csv", "null_string_result": None}, id="with-csv"),
            pytest.param({"file_format": "parquet", "null_string_result": "None"}, id="with-parquet"),
        ],
    )
    def test_fix_dtypes(self, params):
```

File: tests/providers/amazon/aws/transfers/test_sftp_to_s3.py [hunk 2]
```diff
@@ -95,9 +85,10 @@ def test_sftp_to_s3_operation(self, use_temp_file=True):
         create_file_task.execute(None)
 
         # Test for creation of s3 bucket
+        s3_hook = S3Hook("aws_default")
```
Comment (Taragolis) [inline tests/providers/amazon/aws/transfers/test_sftp_to_s3.py]: We do not need any connection here we have mock s3 anyway and better use as keyword
```suggestion
        s3_hook = S3Hook(aws_conn_id=None)
```

File: tests/providers/amazon/aws/transfers/test_s3_to_sftp.py [hunk 3]
```diff
@@ -84,7 +79,8 @@ def test_s3_to_sftp_operation(self):
 
         with open(LOCAL_FILE_PATH, "w") as file:
             file.write(test_remote_file_content)
-        self.s3_hook.load_file(LOCAL_FILE_PATH, self.s3_key, bucket_name=BUCKET)
+        s3_hook = S3Hook("aws_default")
```
Comment (Taragolis) [inline tests/providers/amazon/aws/transfers/test_s3_to_sftp.py]:
```suggestion
        s3_hook = S3Hook(aws_conn_id=None)
```

File: tests/providers/amazon/aws/transfers/test_dynamodb_to_s3.py [hunk 4]
```diff
@@ -18,13 +18,12 @@
...
 import json
-import unittest
 from unittest.mock import MagicMock, patch
 
 from airflow.providers.amazon.aws.transfers.dynamodb_to_s3 import DynamoDBToS3Operator
 
 
-class DynamodbToS3Test(unittest.TestCase):
+class DynamodbToS3Test:
     def setUp(self):
```
Comment (Taragolis) [inline tests/providers/amazon/aws/transfers/test_dynamodb_to_s3.py]: For some reason keeping `setUp` not affect test execution `¯\_(ツ)_/¯`

```suggestion
class DynamodbToS3Test:
    def setup_method(self):
```

Comment (IAL32) [reply tests/providers/amazon/aws/transfers/test_dynamodb_to_s3.py]: I made the change anyways, wouldn't want that to bite us in the back later on ahah

File: tests/providers/amazon/aws/sensors/test_batch.py [hunk 5]
```diff
@@ -42,7 +42,7 @@ def setup_method(self):
     @mock.patch.object(BatchClientHook, "get_job_description")
     def test_poke_on_success_state(self, mock_get_job_description):
         mock_get_job_description.return_value = {"status": "SUCCEEDED"}
-        assert self.batch_sensor.poke({})
+        assert self.batch_sensor.poke({}) is True
```
Comment (IAL32) [inline tests/providers/amazon/aws/sensors/test_batch.py:45]: Small nitpick from @ferruzzi :
https://github.com/apache/airflow/pull/28139#pullrequestreview-1207017465

Comment (ferruzzi) [reply tests/providers/amazon/aws/sensors/test_batch.py:45]: Thanks for including it here.  :+1:

**Review by ferruzzi — APPROVED** [review]
LGTM.  These are a big help, thanks for taking them on!

**Review by Taragolis — APPROVED** [review]

**Pull request merged**